proxy/start_container_interceptor.go

```diff
@@ -1,175 +1,26 @@
 package proxy
 
 import (
-	"bufio"
-	"bytes"
-	"errors"
-	"fmt"
-	"io/ioutil"
-	"net"
 	"net/http"
-	"os"
-	"sort"
-	"strings"
-
-	"github.com/fsouza/go-dockerclient"
-	. "github.com/weaveworks/weave/common"
 )
 
 type startContainerInterceptor struct{ proxy *Proxy }
 
 func (i *startContainerInterceptor) InterceptRequest(r *http.Request) error {
-	return nil
+	container, err := inspectContainerInPath(i.proxy.client, r.URL.Path)
+	if err == nil && containerShouldAttach(container) {
+		i.proxy.createWait(container.ID)
+	}
+	return err
 }
 
 func (i *startContainerInterceptor) InterceptResponse(r *http.Response) error {
-	if r.StatusCode < 200 || r.StatusCode >= 300 { // Docker didn't do the start
+	if r.StatusCode != 201 && r.StatusCode != 204 { // Docker didn't do the start
```

Comment:
This is already in with:
 `if r.StatusCode < 200 || r.StatusCode >= 300 { // Docker didn't do the start`